src/scripts/generate-sdk.ts

```diff
@@ -1,37 +1,174 @@
+  paths?: {
+    [path: string]: {
+      [method: string]: {
+        operationId?: string;
+      };
+    };
+  };
+};
+
+function generateOperationIdMappings(
+  oldSpec: BasicOpenAPISpec,
+  newSpec: BasicOpenAPISpec,
+): Record<string, string> {
+  const mappings: Record<string, string> = {};
+
+  if (newSpec.paths && oldSpec.paths) {
+    for (const [path, pathItem] of Object.entries(newSpec.paths)) {
+      for (const [method, operation] of Object.entries(pathItem)) {
+        const oldOperation = oldSpec.paths[path]?.[method];
+        if (
+          operation.operationId &&
+          oldOperation?.operationId &&
+          operation.operationId !== oldOperation.operationId
+        ) {
+          mappings[operation.operationId] = oldOperation.operationId;
+        }
+      }
+    }
+  }
+
+  console.log("Operation ID mappings:", mappings);
+  return mappings;
+}
+
+function applyOperationIdMappings(
+  originalCode: string,
+  fileName: string,
+  mappings: Record<string, string>,
+): string {
+  const replacementLog: string[] = [];
+  let newCode: string = originalCode;
+
+  for (const [newId, oldId] of Object.entries(mappings)) {
+    const regex: RegExp = new RegExp(`public\\s+${newId}\\(`, "g");
+    const methods = newCode.match(regex);
+
+    if (methods) {
+      for (const method of methods) {
+        const newMethod = `public ${oldId}(`;
+        if (newMethod !== method) {
+          newCode = newCode.replace(method, newMethod);
+          replacementLog.push(
+            `In ${fileName}: Replaced "${method}" with "${newMethod}"`,
+          );
+        }
+      }
+    }
+  }
+
+  fs.appendFileSync(REPLACE_LOG_FILE, `${replacementLog.join("\n")}\n`);
+  return newCode;
+}
+
+function processErcServices(
+  originalCode: string,
+  tag: string,
+  fileName: string,
+): string {
+  const replacementLog: string[] = [];
+  let newCode: string = originalCode;
+
+  const regex: RegExp = new RegExp(`public\\s+${tag}(\\w+)\\(`, "g");
+  const methods = newCode.match(regex);
+
+  if (methods) {
+    for (const method of methods) {
+      const newMethod = method.replace(
+        regex,
+        (_, p1) => `public ${p1.charAt(0).toLowerCase() + p1.slice(1)}(`,
+      );
+      if (newMethod !== method) {
+        newCode = newCode.replace(method, newMethod);
+        replacementLog.push(
+          `In ${fileName}: Replaced "${method}" with "${newMethod}"`,
+        );
+      }
+    }
+  }
+
+  fs.appendFileSync(REPLACE_LOG_FILE, `${replacementLog.join("\n")}\n`);
+  return newCode;
+}
+
+async function main(): Promise<void> {
   try {
-    const response = await fetch(ENGINE_OPENAPI_URL);
-    const jsonData = await response.json();
-
-    // Save the JSON response to a file
-    fs.writeFileSync(
-      "openapi.json",
-      JSON.stringify(jsonData, null, 2),
-      "utf-8",
+    fs.writeFileSync(REPLACE_LOG_FILE, "");
+
+    const oldSpec = JSON.parse(
+      fs.readFileSync("sdk/old_openapi.json", "utf-8"),
```

Comment:
Yeah don't think we need this if we just checkin the SDK src moving forward

Reply:
We would still need the older schema to generate the mapping right?